### DISCUSSION TITLE
Is Capacitor mature enough for release?

### DISCUSSION THREAD

**Discussion by dzineer** [ionic-team/capacitor#5848] [Q&A] [answered]
I am seriously wondering if Capacitor is mature enough for being released to the public? The point of using frameworks is so that you can rely on those instead of having to develop IOS and Android apps through XCode and Android Studio. When you are forced to debug inside XCode or Android Studio, now you are being led down the path of coding in XCode and Android Studio. Also, why does Capacitor show the example for Angular, ReactJS, and JavaScript but not Vue? Seriously developers have to be pulling their hair out over getting things to work in both platforms, where Ionic Framework is supposed to handle those for you already?!!!

I am using a Macbook Pro 2019 Catalina 10.15.6 with XCode and Android Studio. Where are the compatibility charts that list what things work and don't work and where they work and where they don't work and what versions of each is needed to build a successfully IOS and Android app?

**Comment by mlynch** [comment, marked as answer]
Yes, Capacitor is mature and is being used successfully by many small to large companies in production.

> The point of using frameworks is so that you can rely on those instead of having to develop IOS and Android apps through XCode and Android Studio. When you are forced to debug inside XCode or Android Studio, now you are being led down the path of coding in XCode and Android Studio

This was never the "point" of Capacitor. The point of Capacitor is that you can build 95%+ of your app in cross-platform, standardized web technology, hopefully with lots of code sharing with your web app, but still be able to drop down to native for that last 5% so you can do anything you need feature-wise. We want you to be very productive but also never get stuck because you have full access to every native SDK and tool that regular native devs do, too (you just built your app considerably faster than they did 😅).

In fact, one of my original design decisions was that you would use Xcode/Android Studio for the last mile run and debug. That was deliberate, because so many people were struggling with Cordova CLI tools that abstracted that stuff away and failed to work reliably. Today, of course, we have reliable CLI tooling like cap run so you can avoid this if you want.

Ultimately, official tooling for iOS and Android is required in order to actually interact with Apple proprietary devices and Google tooling, there's literally no way around that in mobile. Whether we hide it from you in a CLI or expose it to you directly in Xcode/Android Studio is purely a UX concern.

> Also, why does Capacitor show the example for Angular, ReactJS, and JavaScript but not Vue?

Can you show me what you're looking at? Capacitor is framework agnostic and doesn't care what you use to build your app. Are you referring to Ionic Framework? Ionic Framework is a UI layer for mobile-focused web apps, it's not the same as Capacitor but would be used in tandem with Capacitor. If you're talking about Ionic Framework that is a whole separate discussion since the amount of work required to add full examples for new frameworks is high and it takes us time to do that.

> Where are the compatibility charts that list what things work and don't work and where they work and where they don't work and what versions of each is needed to build a successfully IOS and Android app? 

https://capacitorjs.com/docs/getting-started/faqs#what-platforms-does-capacitor-support
